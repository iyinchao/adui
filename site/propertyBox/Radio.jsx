import React from "react"
import styles from "./components/propertyBox.scss"
import CodeBox from "./components/CodeBox"
import { Button, Switch, Radio } from "../../components"
import { SIZES } from "./consts"

export default class PropertyBox extends React.Component {
  state = {
    disabled: false,
    size: "small",
  }

  render() {
    const { disabled, size } = this.state

    const codes = `<Radio.Group
  defaultValue={0}
  ${disabled ? "disabled" : ""}
  ${size === "small" ? "" : `size="${size}"`}
>
  <Radio value={1}>单选</Radio>
  <Radio value={2}>单选</Radio>
  <Radio value={3}>单选</Radio>
</Radio.Group>`

    return (
      <div className={styles.wrapper}>
        <div className={styles.property}>
          <div className={styles.left}>
            <Radio.Group defaultValue={1} {...this.state}>
              <Radio value={1}>单选</Radio>
              <Radio value={2}>单选</Radio>
              <Radio value={3}>单选</Radio>
            </Radio.Group>
          </div>
          <div className={styles.right}>
            <div className={styles.rightTitle}>可配置项</div>
            <dl className={styles.controls}>
              <dt className={styles.controlTitle}>尺寸</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {SIZES.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === size}
                      onClick={() => this.setState({ size: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="禁用态"
                  unCheckedText="禁用态"
                  checked={disabled}
                  onChange={(param) => this.setState({ disabled: param })}
                />
              </div>
            </dl>
          </div>
        </div>
        <CodeBox codes={codes} />
      </div>
    )
  }
}
