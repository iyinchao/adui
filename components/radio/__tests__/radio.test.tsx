import React, { useState } from "react"
import { mount } from "enzyme"
import TestRenderer from "react-test-renderer"
import Radio from ".."

const RadioMounter = () => {
  const [checked, setChecked] = useState(false)
  return <Radio checked={checked} onChange={(bool) => setChecked(bool)} />
}

describe("Radio", () => {
  it("Radio 是否正确渲染", () => {
    const wrapper = TestRenderer.create(<Radio>children</Radio>)
    expect(wrapper).toMatchSnapshot()
  })

  it("Radio 是否能内部驱动", () => {
    const onChange = jest.fn()
    const wrapper = mount(<Radio onChange={onChange} />)
    wrapper.find("label").at(0).simulate("click")
    expect(
      wrapper.find("label").at(0).props().className.includes("checked")
    ).toEqual(true)
    expect(onChange).toHaveBeenCalledWith(true)
  })

  it("Radio 是否能受外部控制", () => {
    const onChange = jest.fn()
    const wrapper0 = mount(<Radio checked={false} onChange={onChange} />)
    const wrapper1 = mount(<RadioMounter />)
    /**
     * 1. 确定 onChange 还是会传回应该变化的值；
     * 2. 但是由于 Radio 完全受外部控制，所以要确定内部状态不会改变。
     */
    wrapper0.find("label").at(0).simulate("click")
    expect(onChange).toHaveBeenCalledWith(true)
    expect(
      wrapper0.find("label").at(0).props().className.includes("checked")
    ).toEqual(false)

    wrapper1.find("label").at(0).simulate("click")
    expect(
      wrapper1.find("label").at(0).props().className.includes("checked")
    ).toEqual(true)
    wrapper1.find("label").at(0).simulate("click")
    expect(
      wrapper1.find("label").at(0).props().className.includes("checked")
    ).toEqual(true)
  })

  it("Radio 是否能响应 onClick", () => {
    const onClick = jest.fn()
    const wrapper = mount(<Radio onClick={onClick} />)
    wrapper.find("label").at(0).simulate("click")
    expect(onClick).toHaveBeenCalled()
  })

  it("Radio 是否能响应 enter", () => {
    const onChange = jest.fn()
    const wrapper = mount(<Radio onChange={onChange} />)
    wrapper
      .find("label")
      .at(0)
      .simulate("keydown", { key: "enter", keyCode: 13 })
    expect(
      wrapper.find("label").at(0).props().className.includes("checked")
    ).toEqual(true)
    expect(onChange).toHaveBeenCalledWith(true)
  })
})
