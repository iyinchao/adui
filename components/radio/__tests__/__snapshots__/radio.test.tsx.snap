// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Radio 是否正确渲染 1`] = `
<label
  aria-checked={false}
  className="adui-radio-base adui-radio-small"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="radio"
>
  <span
    className="adui-radio-indicator"
  />
  <span>
    children
  </span>
</label>
`;
